pkg/e2e/verify/namespace_webhook.go

```diff
@@ -159,6 +163,80 @@ var _ = ginkgo.Describe(namespaceWebhookTestName, func() {
+			}
+		}, viper.GetFloat64(config.Tests.PollingTimeout))
+
+		util.GinkgoIt("Non-privileged users cannot create prometheusrule in privileged namespaces", func() {
+			for privilegedNamespace := range PRIVILEGED_NAMESPACES {
+				prome2etestrule := e2eCreatePrometheusRule(prometheusName, privilegedNamespace, DUMMY_USER, DUMMY_GROUP, h)
+				err := wait.PollImmediate(5*time.Second, 3*time.Minute, func() (bool, error) {
+					_, err := h.Prometheusop().MonitoringV1().PrometheusRules(privilegedNamespace).Create(context.TODO(), prome2etestrule, metav1.CreateOptions{})
+					if err != nil {
+						log.Printf("failed creating prometheus rules: %v", err)
+						return false, nil
+					}
+					return true, nil
+				})
+				Expect(err).To(HaveOccurred(), "Could  create PrometheusRules")
+			}
+		}, viper.GetFloat64(config.Tests.PollingTimeout))
+
+		util.GinkgoIt("Privileged users can create prometheusrule in all namespaces", func() {
+			for _, privilegedUser := range PRIVILEGED_USERS {
+				for privilegedNamespace := range PRIVILEGED_NAMESPACES {
+					prome2etestrule := e2eCreatePrometheusRule(prometheusName, privilegedNamespace, privilegedUser, "", h)
+					err := wait.PollImmediate(5*time.Second, 3*time.Minute, func() (bool, error) {
+						_, err := h.Prometheusop().MonitoringV1().PrometheusRules(privilegedNamespace).Create(context.TODO(), prome2etestrule, metav1.CreateOptions{})
+						if err != nil {
+							log.Printf("failed creating prometheus rules: %v", err)
+							return false, nil
+						}
+						return true, nil
+					})
+					Expect(err).To(HaveOccurred())
+				}
+				for _, namespace := range NONPRIV_NAMESPACES {
+					prome2etestrule := e2eCreatePrometheusRule(prometheusName, namespace, privilegedUser, "", h)
+					err := wait.PollImmediate(5*time.Second, 3*time.Minute, func() (bool, error) {
+						_, err := h.Prometheusop().MonitoringV1().PrometheusRules(namespace).Create(context.TODO(), prome2etestrule, metav1.CreateOptions{})
+						if err != nil {
+							log.Printf("failed creating prometheus rules: %v", err)
+							return false, nil
+						}
+						return true, nil
+					})
+					Expect(err).To(HaveOccurred())
+				}
+			}
+		}, viper.GetFloat64(config.Tests.PollingTimeout))
+
+		util.GinkgoIt("Non-privileged users can create prometheusrule all non-privileged namespaces", func() {
+			for _, nonPrivilegedNamespace := range NONPRIV_NAMESPACES {
+				prome2etestrule := e2eCreatePrometheusRule(prometheusName, nonPrivilegedNamespace, DUMMY_USER, "dedicated-admins", h)
+				err := wait.PollImmediate(5*time.Second, 3*time.Minute, func() (bool, error) {
+					_, err := h.Prometheusop().MonitoringV1().PrometheusRules(nonPrivilegedNamespace).Create(context.TODO(), prome2etestrule, metav1.CreateOptions{})
+					if err != nil {
+						log.Printf("failed creating prometheus rules: %v", err)
+						return false, nil
+					}
+					return true, nil
+				})
+				Expect(err).To(HaveOccurred())
```

Comment:
This block can go in `e2eCreatePrometheusRule` to de-duplicate all of the tests doing this just on a different namespace (no need to return the prometheusrule)